plugins/org.jkiss.dbeaver.model.jdbc/src/org/jkiss/dbeaver/model/impl/jdbc/JDBCDataSource.java

```diff
@@ -272,6 +272,9 @@ protected Connection openConnection(@NotNull DBRProgressMonitor monitor, @Nullab
                 }
             };
 
+            ClassLoader originalClassLoader = Thread.currentThread().getContextClassLoader();
+            Thread.currentThread().setContextClassLoader(driver.getClassLoader());
```

Comment:
We dynamically load some classes using `ServiceLoader` in ScalarDB Core and SQL. But, maybe due to the plugin framework of CloudBeaver, the context class loader is `org.eclipse.osgi.internal.loader.EquinoxClassLoader` and it cannot find target classes outside the driver Jar. Fortunately, `DBPDriver` knows where those are, so change the class loader temporarily.